src/subcommand/status_subcommand.cpp

```diff
@@ -290,4 +300,21 @@ void status_subcommand::run()
     //         std::cout << std::endl;
     //     }
     // }
+
+    if (!sl.has_tobecommited_header() & (sl.has_notstagged_header() | sl.has_untracked_header()))
```

Comment:
I think these should be `&&` and `||` (logical operators) rather than `&` and `|` (bitwise operators).